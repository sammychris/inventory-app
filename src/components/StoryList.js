import React from 'react';
import styled from 'styled-components';
import Button from './Button';


const StoryList = ({stories, onUpdateStory, onDeleteStory}) => {
    return (
        stories.map((item, i) => {
            const {id, story} = item; 
            return (
                <ItemContainer key={i}>
                    <span>Id: {id} </span><span>Story: {story}</span>
                    <Button event={() => onUpdateStory(item)}>Edit</Button>
                    <Button event={() => onDeleteStory(item)}>Delete</Button>
                </ItemContainer>
            )
        })
    );
  }


export default StoryList;

const ItemContainer = styled.div`
  color: grey;
  font-weight: bolder;
  padding: 10px;
  border-color: #cfcfcf;
  border-width: 0.1em 0;
  border-style: solid;
  &:hover {
    box-shadow: 1px 3px 1px 1px #f5f5f5;
    z-index: 1;
    border-width: 1px;
  }

`;
