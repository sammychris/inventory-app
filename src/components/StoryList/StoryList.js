import React from 'react';
import styled from 'styled-components';
import Button from '../Button';


const StoryList = ({stories, onUpdateStory, onDeleteStory}) => {
    if(!stories.length) return <Message>No story yet!</Message>;
    return (
        stories.map((item, i) => {
            const {id, story} = item; 
            return (
                <ItemContainer key={i}>
                    <Span>id-{id}</Span><Span>{story}</Span>
                    <div>
                      <Button cls="primary" event={() => onUpdateStory(item)}>Edit</Button>
                      <Button cls="danger" event={() => onDeleteStory(item)}>Delete</Button>
                    </div>
                </ItemContainer>
            )
        })
    );
  }


export default StoryList;

const ItemContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items:center;
  padding: 0 10px;
  border-color: #cfcfcf;
  border-width: 0.1em 0;
  border-style: solid;
  &:hover {
    box-shadow: 1px 3px 1px 1px #f5f5f5;
    z-index: 1;
    border-width: 1px;
  }
`;

const Span = styled.span`
  color: cadetblue;
  font-size: 16px;
  font-weight: 600;
`;

const Message = styled.div`
  margin-top: 20px;
  text-align: center;
`;
