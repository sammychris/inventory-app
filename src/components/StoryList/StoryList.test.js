import { shallow } from "enzyme";
import StoryList from "./StoryList";

test("expect to render a StoryList", () => {
  const props = { 
    stories: [{}],
    onUpdateStory: jest.fn(),
    onDeleteStory: jest.fn(),
  };
  const wrapper = shallow(<StoryList {...props} />);
  expect(wrapper.length).toEqual(1);
});

test("expect to make a snap shot for StoryList component", () => {
  const props = {
    stories: [{}],
    onUpdateStory: jest.fn(),
    onDeleteStory: jest.fn(), 
  };
  const wrapper = shallow(<StoryList {...props} />);
  expect(wrapper).toMatchSnapshot();
});