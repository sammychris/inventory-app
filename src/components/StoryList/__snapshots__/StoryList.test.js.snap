// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to make a snap shot for StoryList component 1`] = `ShallowWrapper {}`;
